adapters/resetdigital/resetdigital.go

```diff
@@ -0,0 +1,284 @@
+		resetDigitalRequestData.Imps[0].ForceBid = true
+	}
+
+	return resetDigitalRequestData, nil
+}
+
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent || response.StatusCode == http.StatusBadRequest || response.StatusCode != http.StatusOK {
+		return nil, nil
+	}
+
+	var bidResp map[string]interface{}
+	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(maxBids)
+	jsonData := bidResp["bids"].([]interface{})
+
+	for _, bidData := range jsonData {
+		bidMap := bidData.(map[string]interface{})
+		bid := getBidFromResponse(bidMap)
+		if bid == nil {
+			continue
+		}
+
+		bidTypes, err := getBidTypes(internalRequest.Imp[0])
+		if err != nil {
+			return nil, []error{err}
+		}
+
+		for _, bidType := range bidTypes {
+			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+				Bid:     bid,
+				BidType: bidType,
+			})
+		}
+	}
+
+	bidResponse.Currency = getCurrency(internalRequest)
+	return bidResponse, nil
+}
+func getBidFromResponse(requestData map[string]interface{}) *openrtb2.Bid {
+	processData := requestData["bid_id"].(string)
+
+	bid := &openrtb2.Bid{
+		ID:    processData,
+		Price: getBidPrice(requestData),
+		ImpID: requestData["imp_id"].(string),
+		CrID:  requestData["crid"].(string),
+	}
+
+	if value, ok := requestData["html"].(string); ok {
+		bid.AdM = value
+	}
+
+	if value, ok := requestData["w"].(string); ok {
+		if i, err := strconv.ParseInt(value, 10, 64); err == nil && i > 0 {
+			bid.W = i
+		}
+	}
+
+	if value, ok := requestData["h"].(string); ok {
+		if i, err := strconv.ParseInt(value, 10, 64); err == nil && i > 0 {
+			bid.H = i
+		}
+	}
+
+	if bid.Price == 0 {
+		return nil
+	}
+
+	return bid
+}
+
+func getBidPrice(requestData map[string]interface{}) float64 {
+	if value, ok := requestData["cpm"].(float64); ok {
+		return value
+	}
+	return 0.0 // Default value if "cpm" doesn't exist or is not a float64
+}
+
+func getBidType(imp openrtb2.Imp) (openrtb_ext.BidType, error) {
+	if imp.Banner != nil {
+		return openrtb_ext.BidTypeBanner, nil
+	} else if imp.Video != nil {
+		return openrtb_ext.BidTypeVideo, nil
+	} else if imp.Audio != nil {
+		return openrtb_ext.BidTypeAudio, nil
+	}
+
+	return "", fmt.Errorf("failed to find matching imp for bid %s", imp.ID)
+}
+
+func getBidTypes(imp openrtb2.Imp) ([]openrtb_ext.BidType, error) {
+	var bidTypes []openrtb_ext.BidType
+
+	if imp.Banner != nil {
+		bidTypes = append(bidTypes, openrtb_ext.BidTypeBanner)
+	}
+	if imp.Video != nil {
+		bidTypes = append(bidTypes, openrtb_ext.BidTypeVideo)
+	}
+	if imp.Audio != nil {
+		bidTypes = append(bidTypes, openrtb_ext.BidTypeAudio)
+	}
+	if len(bidTypes) == 0 {
+		return nil, fmt.Errorf("failed to find matching imp for bid %s", imp.ID)
+	}
```

Comment:
https://github.com/prebid/prebid-server/pull/3766#discussion_r1671276804 this comment mentions like multi format bids are not supported. But `getBidTypes` collects all multiple format available for bid.